products/bluebell/core/src/passes/state_allocator.rs

```diff
@@ -0,0 +1,226 @@
+use crate::intermediate_representation::symbol_table::StateLayoutEntry;
+use primitive_types::U256;
+
+use crate::constants::{TraversalResult, TreeTraversalMode};
+use crate::intermediate_representation::pass::IrPass;
+use crate::intermediate_representation::pass_executor::PassExecutor;
+use crate::intermediate_representation::primitives::CaseClause;
+use crate::intermediate_representation::primitives::ContractField;
+use crate::intermediate_representation::primitives::Instruction;
+use crate::intermediate_representation::primitives::{
+    ConcreteFunction, ConcreteType, EnumValue, FunctionBlock, FunctionBody, FunctionKind,
+    IntermediateRepresentation, IrIdentifier, IrIndentifierKind, Operation, Tuple,
+    VariableDeclaration, Variant,
+};
+use crate::intermediate_representation::symbol_table::SymbolTable;
+use log::info;
+
+pub struct StateCollector {
+    namespace_stack: Vec<String>,
+    current_namespace: Option<String>,
+    current_type: Option<String>,
+    address_offset: u64,
+}
+
+impl StateCollector {
+    pub fn new() -> Self {
+        StateCollector {
+            namespace_stack: Vec::new(),
+            current_namespace: None,
+            current_type: None,
+            address_offset: 4919, // TODO:
```

Comment:
Magic number?

Reply:
Yeah, it translates into 1337 in hex which makes it easy to see that whether you are writing to the correct address or not. It needs to be replaced by the keccak of the name though.